x/ecocredit/core/msg_credit_type.go

```diff
@@ -0,0 +1,21 @@
+package core
+
+import (
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+func (m CreditType) Validate() error {
+	if err := ValidateCreditTypeAbbreviation(m.Abbreviation); err != nil {
+		return err
+	}
+	if len(m.Name) == 0 {
+		return sdkerrors.ErrInvalidRequest.Wrap("name cannot be empty")
+	}
+	if len(m.Unit) == 0 {
+		return sdkerrors.ErrInvalidRequest.Wrap("unit cannot be empty")
+	}
```

Comment:
I think #429 would be nice to have alongside these updates but this could be added as a followup.